### ISSUE TITLE
Encryption in the client : All the files were decrypted without the need for me to enter my password

### ISSUE THREAD

**Issue by Lattitude75** [laurent22/joplin#1591]
### Operating system

- Android

### Application
- Mobile

I enabled the E2E encryption on my desktop and synchronised my notes with the cloud. On my client, I gave Joplin access to my cloud account for synchronisation and found that all the notes were decrypted and loaded without the need for me to enter my password. I do not know if this is a bug or it is supposed to work this way. 

If this is the case, the client only needs access to the drive to get the notes decrypted without needing a password. Then there is no point in encryption right? I was expecting that the synchronisation will leave me with a set of encrypted files and the notes can only be recovered after entering the password. I hope that this will be corrected in a future release.

**Comment by CalebJohn** [comment]
Is it possible that your notes had already synced un-encrypted and that's what you pulled down? It should be impossible for the client to un-encrypt notes without the password. 

Could you double check the encryption settings to know that they're set up correctly?

If the notes aren't encrypting correctly, that's a huge issue.

**Comment by Lattitude75** [comment]
No that is not the case. To make sure, I have checked the files on my Dropbox and found that all of them were encrypted. To further verify it, I resynced my notes after enabling the encryption on the desktop, deleted the app data for Joplin on Android and synchronised the notes. Now, I see that the master key is received, and the files received have also been decrypted without entering the password.

**Comment by tessus** [comment]
Hmm, you said you deleted the app data, but did you close Joplin on Android? If not, the decryption password was still in the app memory. You did not delete the app itself, did you?

**Comment by Lattitude75** [comment]
Yeah, the app closes itself once you delete the app data. I will try uninstalling the application and re-synchronise the notes.

**Comment by tessus** [comment]
Please send @laurent22 directly an email for security related issues. They should not be disclosed publicly. This was stated on the web site at one point, but it seems this info is gone.
@laurent22 we should add this back to the web site that security related issues must not be opened in github (we can't delete issues or restrict them to the reporter and team).

**Comment by laurent22** [comment]
Emailing me directly is just for security issues that can are understood and can be replicated (as those done by security researchers) but this looks like a different issue. Obviously the app cannot guess the password and decrypt data by itself so there's something else going on. Either the data was not encrypted on the target or the password was in the database cache.

**Comment by tessus** [comment]
At least a DM in discourse should be used. I find it rather strange to discuss a security issue in public, but it's your decision.

**Comment by laurent22** [comment]
Discussing a replicable security exploit in public we should avoid, but this seems to be a setup issue. @Lattitude75 could you check the following to confirm?

- Is all data really encrypted on the sync target? How did you check? An encrypted item will have the field "encryption_applied" set to "1".
- On desktop, could you open the Encryption config page? In there, is the password set for the  master key? (is there a tick next to it?)

**Comment by Lattitude75** [comment]
I checked the files in the resource folder on the cloud. I found that they were randomly named and I couldn't open them, I thus assumed that my data was encrypted. I now see that the .md files are not encrypted and the flag encryption_applied is 0.

While, the encryption config on the desktop app shows that encryption is enabled and has a green tick on the password of the master key. But, I am sure that I synchronised my cloud data at least thrice and the first sync took a long time. I will try to re sync my notes with a new password and this time, I will try to start by deleting the data on cloud, to make sure that any non encrypted data is removed. I will get back to you on what I'll find.

**Comment by tessus** [comment]
@Lattitude75 backup your notes by exporting them. if you delete the data on the sync target, your local data will be deleted as well.

**Comment by laurent22** [comment]
Also please have a look at the E2EE guide: https://joplinapp.org/e2ee/

If you already have notes on the sync target, it takes a while because everything needs to be encrypted and re-uploaded.

**Comment by laurent22** [comment]
By the way, to check if the files on a sync target are encrypted, you can use the `joplin` command line tool. To do so:

1. Copy all the files from your sync target (Nextcloud, Dropbox, etc.) to a local folder on your computer
2. Run `joplin e2ee target-status /path/to/yourfiles`

That should output something like this:

```
Encrypted items: 6595/6595
Encrypted resources: 750/750
Other items (never encrypted): 1
```

The "never encrypted" item is the master key, which content is encrypted differently, with the user password.

**Comment by Lattitude75** [comment]
@laurent22 I have followed the guide or E2E encryption every time. Today, I deleted all the configuration files for Joplin in desktop and the Android device and started the setup again. This time, the encryption was successful and I was asked for a password to recover notes for the client. Taking a backup helped as I had to recover notes in a fresh installation. Seems like, it didn't synchronise well the first time.
